utils/artemis_utils/service.py

```diff
@@ -14,28 +15,44 @@
 def get_local_ip():
     return socket.gethostbyname(socket.gethostname())
 
+def resolve_dns(query:str, rtype:str = 'A', timeout:int = 2)->list:
+  rtype.upper()
+  resolver = dns.resolver.Resolver()
+  if rtype == "PTR":
+    query = dns.reversename.from_address(query)
+  msg = dns.message.make_query(query,rtype)
+  for dns_server in resolver.nameservers:
+    try:
+      resp = dns.query.udp(msg,dns_server,timeout=timeout)
+      if resp.answer:
+        return [str(a) for a in resp.answer[0] ]
+    except Exception as e:
+       log.error("error:",dns_server, e)
+  return []
+
 
 def service_to_ips_and_replicas_in_compose(own_service_name, base_service_name):
     local_ip = get_local_ip()
+    address_regexp = re.compile ('\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}')
```

Comment:
what about IPv6 though?

Reply:
Hi. good point. i have amended resolve_dns to lookup AAAA and A by default. Oddly, podman's DNS seems to respond  with an 'A' even if query type is 'AAAA' and 'AAAA' does not exist. Docker does not act in this way. For this reason, we dedupe the response before returning. 

dig -t AAAA database
;; QUESTION SECTION:
;database.			IN	AAAA 
;; ANSWER SECTION:
database.		86400	IN	A	10.89.0.85


tested this code on docker and podman